packages/bbui/src/Form/Core/Checkbox.svelte

```diff
@@ -29,6 +30,11 @@
     checked={value}
     {disabled}
     on:change={onChange}
+    on:click={e => {
+      if (readonly) {
+        e.preventDefault()
+      }
+    }}
```

Comment:
Rather than hijacking clicks, for these components that don't natively support readonly, could we use a CSS solution instead?

I'm thinking something like:
```js
<label
  class="spectrum-Checkbox spectrum-Checkbox--emphasized {sizeClass}"
  class:readonly
>
....
<style>
.readonly {
  pointer-events: none;
}
</style>
```

That's more performant in the sense that it reduces additional event handlers, and also avoids any potential side effects to due handling click events.

Reply:
https://github.com/Budibase/budibase/assets/101575380/2abf6732-ec42-49fc-bd9e-55ba5242cd10

I gave this a go, but I ran into this problem were the click event wasn't prevented on the label. Not sure why.

Reply:
As discussed, added the readonly class to the parent Field component label.